python/runtime/db_test.py

```diff
@@ -402,5 +403,51 @@ def test_get_table_schema(self):
             self.assertTrue(np.array_equal(expect, schema))
 
 
+class TestMySQLFieldType(TestCase):
+    @unittest.skipUnless(
+        os.getenv("SQLFLOW_TEST_DB") == "mysql", "run only in mysql")
+    def test_field_type(self):
+        self.assertGreater(len(MYSQL_DATA_TYPE_DICT), 0)
+
+        addr = os.getenv("SQLFLOW_TEST_DB_MYSQL_ADDR", "localhost:3306")
+        conn = connect_with_data_source(
+            "mysql://root:root@tcp(%s)/?maxAllowedPacket=0" % addr)
+        cursor = conn.cursor()
+
+        table_name = "iris.test_mysql_field_type_table"
+        drop_table_sql = "DROP TABLE IF EXISTS %s" % table_name
+        create_table_sql = "CREATE TABLE IF NOT EXISTS " + table_name + "(a %s)"
+        select_sql = "SELECT * FROM %s" % table_name
+
+        for int_type, str_type in MYSQL_DATA_TYPE_DICT.items():
+            if str_type in ["VARCHAR", "CHAR"]:
+                str_type += "(255)"
+
+            cursor.execute(drop_table_sql)
+            cursor.execute(create_table_sql % str_type)
+            cursor.execute(select_sql)
+
+            int_type_actual = cursor.description[0][1]
+            cursor.execute(drop_table_sql)
+
+            self.assertEqual(int_type_actual, int_type,
+                             "%s not match" % str_type)
+
+
+class TestLimitSelect(TestCase):
+    def test_limit_select(self):
+        self.assertEqual("SELECT * FROM t LIMIT 2",
+                         limit_select("SELECT * FROM t LIMIT 30", 2))
+
+        self.assertEqual("SELECT * FROM t LIMIT 30; \t",
```

Comment:
30 -> 100?

Reply:
No. It would be 30 here. The `limit_select` method would choose `min(original_limit_num, new_limit_num)` as the new limited number.